Patch 1

HiteshDhola opened the pull request:
Updated procedure after testing it on Windows 10 and few specific points.

HiteshDhola committed:
Update installation-windows.rst

HiteshDhola committed:
Update installation-windows.rst

anjohnson commented on getting-started/installation-windows.rst:
```diff
@@ -90,18 +97,31 @@ Install EPICS
-    softIoc
-    epics>
+    $ /home/Hitesh/base-7.0.3.1/bin/windows-x64-mingw/softIoc
+    $ epics>
```
There should be no `$ ` before the `epics>`, this line shows the expected prompt from the IOC. The `$ ` that you added to the line above is the prompt from bash, which is correct.

It would be better to avoid including personal user-names in these instructions; earlier on this document suggests that Base be installed in $HOME, but I would suggest putting it somewhere like `C:\epics\base-7.0.3.1` might be a good way to avoid that. Alternatively use `$HOME` instead of `/home/Hitesh` in the path shown here.

Oh, and the version number in the path should match the version number of the tarfile that is downloaded, so please either change this path to use 7.0.3 or update the "Install EPICS" section above.

HiteshDhola replied:
1) Name is left there in hurry, everywhere else I avoided it. I will correct it with 'user'. 
2) For path like "c:\epics,,," : in Msys environment paths are like linux, so it starts with /home. 
3) For $HOME : while preparing documentation I tried to simplify for very beginner level users. putting $HOME might confuse some winows user. I will try to avoid name and simplify it.

All other points I would modify as per your suggestion

HiteshDhola replied:
Modification done and comitted.

anjohnson commented on getting-started/installation-windows.rst:
```diff
@@ -90,18 +97,31 @@ Install EPICS
+
+::
+
+     cd c:\msys64\home\Hitesh\base-7.0.3.1\bin\windows-x64-mingw
```
Since you added the bash prompt earlier, shouldn't this example show the cmd.exe prompts to match?

Same comment as above regarding the path to the installation, but I'm not sure that `%HOME%` would be set, I suspect that's something that MSYS may be setting when you run bash, which isn't happening here. That's why using `C:\epics` might be better.

HiteshDhola replied:
I tried to use bash prompt but it is symbol ">", which can get confused with when IOC starts and "epics>" appears. So, I removed it. 

For Path, I would remove name and put 'user' into it. I think putting actual path will help users, as switching between msys and windows bash will require different types of path. So, I put whole paths when using windows bash and relative path in while using Msys. 

I am trying to prepare documentation such that, If any person follows instruction as it is and copies the commands, it should work without any modification.

If I use path like c:\epics, Msys would not have access to that location.

HiteshDhola replied:
Relavent changes made and committed.

anjohnson commented on getting-started/installation-windows.rst:
```diff
@@ -90,18 +97,31 @@ Install EPICS
+     cd c:\msys64\home\Hitesh\base-7.0.3.1\bin\windows-x64-mingw
+     softIoc.exe
+     dbLoadDatabase("c:\msys64\home\Hitesh\base-7.0.3.1\bin\windows-x64-mingw\../../dbd/softIoc.dbd")
+     softIoc_registerRecordDeviceDriver(pdbbase)
```
There's an easier way to test the softIoc without having to type the iocsh commands. The `cd` command isn't really necessary here either since the executable will look in its own bin directory for the DLLs that it needs. This should be all you need to type:

```
C:\>\epics\base-7.0.3.1\bin\windows-x64-mingw\softIoc -x test
Starting iocInit
############################################################################
## EPICS R7.0.3.1
## EPICS Base built Feb 27 2020
############################################################################
iocRun: All initialization complete
epics> dbl
test:BaseVersion
test:exit
epics> exit

C:\>
```

That runs the softIoc, loads 2 records from a built-in database file and then runs iocInit. The parameter to the `-x` flag is a prefix for the records. I also added the `dbl` command at the iocsh prompt to show that those records have been loaded, and ran exit when I was finished.

HiteshDhola replied:
I tried to match "Installation on Linux/UNIX/DARWIN" procedures. So that documentation looks uniform. But I can change to your suggestion, it looks simpler.

HiteshDhola replied:
Updated as advised. Committed.

anjohnson commented on getting-started/installation-windows.rst:
```diff
@@ -66,15 +66,22 @@ Install ``gcc`` for 64-bit environment
     make 4.2.1-1
     perl 5.30.0-1
     mingw-w64-x86_64-gcc 9.2.0-2
+    gcc 9.1.0-2
```
What's the difference between these two copies of gcc? It surprises me slightly that they have two different version numbers, which one is actually being used?

HiteshDhola replied:
If I only install gcc and not mingw-gcc, make fails giving lots of error.
If I only install mingw-gcc and not gcc, make fails whenever "gcc" or "g++" commands are used.

I am also confused. But keeping both version only works. I failing to attach and error output file, I can send you via email if you want to see it.

ralphlange replied:
`gcc` is the native MSYS2 compiler, `mingw-w64-x86_64-gcc` is the 64bit native MinGW compiler tool chain.

While MSYS2 is a subsystem based on cygwin and focuses on providing a Unix-style environment and package manager for Windows, the MinGW tools provide and create Windows programs that are independent from the subsystem used.

For EPICS compilation we want to use the MinGW tool chains (I have 32bit and 64bit working). I am not sure if and what for the MSYS2 gcc is really necessary. Note that the built-in EPICS platform detection script is failing to detect the MinGW toolchains correctly. EPICS_HOST_ARCH has to be set manually. (That should be fixed in base, probably.)

https://www.msys2.org/wiki/MSYS2-introduction/
has a good overview of the MSYS2 / MinGW ecosystem. (Might be useful to link to.)

ralphlange replied:
See https://bugs.launchpad.net/epics-base/+bug/1873406 for the EPICS_HOST_ARCH detection.

anjohnson reviewed (commented):
Hi Hitesh, thanks for contributing to this documentation. I have some questions and suggestions below which I hope you will consider incorporating into this PR.

HiteshDhola committed:
Update installation-windows.rst

HiteshDhola committed:
Update installation-windows.rst

Removed Name "hitesh" from all description
Added section to test softioc with db file
added windows environment setting procedure
added demo ioc creation and testing in windows environment

HiteshDhola reviewed (commented):
Is there any modification required ? I tested it multiple times and everything is working fine as per my tests. If no change is required, we may merge this.

ralphlange commented:
Yes, I'll merge it.
Sorry for dropping the ball. No bad intentions.
